plinth/frontpage.py

```diff
@@ -27,18 +27,31 @@ def get_shortcuts():
     return sorted(shortcuts.values(), key=lambda item: item['label'])
 
 
-def add_shortcut(id, label, url, icon, details=None):
+def add_shortcut(id, label, url, icon, details=None,login_required=True):
     """Add shortcut to front page."""
     if not url:
         url = '?selected={id}'.format(id=id)
-
-    shortcuts[id] = {
-        'id': id,
-        'label': label,
-        'url': url,
-        'icon': icon,
-        'details': details,
-    }
+    
+    apps_login_required = {'deluge', 'openvpn', 'quassel', 'radicale', 'repro', 'roundcube', 'shaarli', 'transmission', 'ttrss', 'xmpp', 'privoxy'}
```

Comment:
One of the primary purposes of having separate modules is that a developer not involved in Plinth development should be able to create a module for it without modifying the code of core Plinth.  With the above approach that can't happen.  The core Plinth code should not having anything specific for a module.

Instead of doing this, in each of the apps that add a shortcut, modify the add_shortcut call to send the value of login_required parameter (when is not the default value) properly.  Use the value sent when creating the shortcut dictionary.